apps/synapse-oauth-signin/src/AppInitializer.tsx

```diff
@@ -6,18 +7,18 @@ import { handleErrorRedirect } from './URLUtils'
 function AppInitializer(
   props: React.PropsWithChildren<Record<string, unknown>>,
 ) {
-  const urlSearchParams = new URLSearchParams(window.location.search)
-  const prompt = urlSearchParams.get('prompt')
+  const [searchParams] = useSearchParams()
```

Comment:
Use react-router's `useSearchParams` hook. This is required since we're using `MemoryRouter` in tests--the `window.location` object provided by JSDOM is no longer modified.